File: src/temp.cpp
library(Rcpp)

cppFunction(
  "NumericVector temp(int n, int m) {
    
    NumericMatrix chain(n, m);
    double calc_mean;
    NumericVector temp;
    
    
    
    chain(0 , _) = rnorm(m);
    
    temp = chain(0 , _);
    
    calc_mean = mean(temp);
    
    // for (int i = 0; i < n; i++) {
    //   temp = chain(i , _);
    //   calc_mean = mean(temp);
    //   chain(i+1 , _) = rnorm(m, calc_mean, 1);
    // }
    
    
    return calc_mean;
  }"
)

File: R/temp.R
library(Rcpp)

cppFunction(
  "NumericMatrix temp(NumericVector obs, int N) {
    
    double obs_ss = mean(obs);
  
    int length_theta = obs.size();
    NumericMatrix chain(N, length_theta);
    NumericVector update(length_theta);
    update = rnorm(length_theta, 1, 1);
  
    chain(0 , _) = update;
    
    
    NumericVector prop(length_theta);
    for (int i = 1; i < (N-1); i++) {
      prop = 
    }
        

    return chain;
  }"
)

temp(rnorm(21,0,1),10)
